### ISSUE TITLE
compiler panic in 'filesystem/ui/bin/browser' in thml5ever v0.12.1

### ISSUE THREAD

**Issue by andrewufrank** [rust-lang/rust#41669]
it is not very clear from the output, what should be reported and what details about my environment you would be interested in. could you automatically produce the output (or a routine to produce the information) such that it could be `cut and paste` in the message. This would likely simplify your triage of the messages:

here the output - 
```
$ make all
#filesystem/bin/dmesg filesystem/bin/info  filesystem/bin/watch
mkdir -p filesystem/ui/bin
RUSTC="/home/frank/redox/rustc.sh" RUSTDOC="./rustdoc.sh" CARGO_INCREMENTAL=1 cargo rustc --manifest-path programs/orbutils/Cargo.toml --bin browser --target x86_64-unknown-redox --release -- -C codegen-units=`nproc` -o filesystem/ui/bin/browser
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#mio:0.6.2
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#redox_event:0.1.0
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#ralloc:1.0.0
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#same-file:0.1.2
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#extra:0.1.0
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#pager:0.1.0
warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#tar:0.4.11
   Compiling image v0.12.4
   Compiling tendril v0.2.3
   Compiling html5ever v0.12.1
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: run with `RUST_BACKTRACE=1` for a backtrace

thread 'rustc' panicked at 'assertion failed: !unsize', /checkout/src/librustc_typeck/check/mod.rs:2212
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:355
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:365
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:549
   5: std::panicking::begin_panic
   6: rustc_typeck::check::FnCtxt::try_index_step
   7: rustc_typeck::check::method::confirm::ConfirmContext::confirm
   8: rustc_typeck::check::method::<impl rustc_typeck::check::FnCtxt<'a, 'gcx, 'tcx>>::lookup_method
   9: rustc_typeck::check::FnCtxt::check_expr_kind
  10: rustc_typeck::check::FnCtxt::check_expr_with_expectation_and_lvalue_pref
  11: rustc_typeck::check::FnCtxt::check_stmt
  12: rustc_typeck::check::FnCtxt::check_block_with_expected::{{closure}}
  13: rustc_typeck::check::FnCtxt::check_block_with_expected
  14: rustc_typeck::check::FnCtxt::check_expr_kind
  15: rustc_typeck::check::FnCtxt::check_expr_with_expectation_and_lvalue_pref
  16: rustc_typeck::check::FnCtxt::check_return_expr
  17: rustc_typeck::check::check_fn
  18: rustc_typeck::check::typeck_tables_of
  19: rustc::ty::maps::<impl rustc::ty::maps::queries::typeck_tables_of<'tcx>>::try_get
  20: rustc::ty::maps::TyCtxtAt::typeck_tables_of
  21: rustc::ty::maps::<impl rustc::ty::context::TyCtxt<'a, 'tcx, 'lcx>>::typeck_tables_of
  22: rustc_typeck::check::typeck_item_bodies
  23: rustc::ty::maps::<impl rustc::ty::maps::queries::typeck_item_bodies<'tcx>>::try_get
  24: rustc::ty::maps::TyCtxtAt::typeck_item_bodies
  25: rustc::ty::maps::<impl rustc::ty::context::TyCtxt<'a, 'tcx, 'lcx>>::typeck_item_bodies
  26: rustc_typeck::check_crate
  27: rustc_driver::driver::phase_3_run_analysis_passes::{{closure}}
  28: rustc_driver::driver::phase_3_run_analysis_passes
  29: rustc_driver::driver::compile_input
  30: rustc_driver::run_compiler
  31: std::panicking::try::do_call
  32: __rust_maybe_catch_panic
             at /checkout/src/libpanic_unwind/lib.rs:98
  33: <F as alloc::boxed::FnBox<A>>::call_box
  34: std::sys::imp::thread::Thread::new::thread_start
             at /checkout/src/liballoc/boxed.rs:658
             at /checkout/src/libstd/sys_common/thread.rs:21
             at /checkout/src/libstd/sys/unix/thread.rs:84
  35: start_thread
  36: __clone

error: Could not compile `image`.
Build failed, waiting for other jobs to finish...
error: build failed
mk/userspace/orbutils.mk:13: recipe for target 'filesystem/ui/bin/browser' failed
make: *** [filesystem/ui/bin/browser] Error 101

```
additional comment:
it would be better advertisement if you could have a repository where the code is clean enough that it installs and new users, like me, could get a working copy quickly to get an impression and to decide if they want to continue. sort of regular releases which are snapshots (most likely) to compile. 
it is an exciting project, but you need to build your base of followers. for this an initial positive experience is a great plus.

**Comment by kennytm** [comment]
Duplicate of #41604, which should be fixed in #41651.

**Comment by arielb1** [comment]
@andrewufrank

Can you give me the output of `rustc --version`?

**Comment by andrewufrank** [comment]
thank you - i am a new to redox. perhaps you indicate in the manual that it is necessary to use 
    rustup update
and it is not automatic when rustup is set once to nightly. 
now it has compiled and make qemu worked. can you point me to some docs about how to use qemu with respect to redox? 
how much more can be done with redox? can i use rust to compile and run programs? other languages? (eg. haskell which can be using the llvm)? 
thanks for the initiative to build an OS based on a safe language!
andrew
